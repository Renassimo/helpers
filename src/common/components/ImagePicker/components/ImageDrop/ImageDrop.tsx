import Typography from '@mui/material/Typography';

import { DropzoneState } from 'react-dropzone';

const ImageDrop = ({ dropzone }: { dropzone: DropzoneState }) => {
  const { getRootProps, getInputProps, isDragAccept } = dropzone;
  const backgroundColor = isDragAccept ? 'blue' : '#fafafa';
  return (
    <div
      {...getRootProps({
        style: {
          padding: '32px',
          backgroundColor,
          border: '1px dashed gray',
          color: 'gray',
          borderRadius: '4px',
        },
      })}
    >
      <input {...getInputProps()} />
      <Typography>Drag and drop map image, or click to select...</Typography>
    </div>
  );
};

export default ImageDrop;
