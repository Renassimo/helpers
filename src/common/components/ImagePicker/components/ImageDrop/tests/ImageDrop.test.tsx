import renderWithTheme from '@/common/tests/helpers/renderWithTheme';

import { DropzoneState } from 'react-dropzone';

import ImageDrop from '../ImageDrop';

describe('ImageDrop', () => {
  const mockedGetRootProps = jest.fn(() => ({
    className: 'mockedGetRootPropsResult',
  }));
  const mockedGetInputProps = jest.fn(() => ({ type: 'input' }));

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('renders snapshot successfully', () => {
    // Act
    const mockedDropzone = {
      getRootProps: mockedGetRootProps,
      getInputProps: mockedGetInputProps,
      isDragAccept: true,
    } as unknown as DropzoneState;
    // Arange
    const { baseElement } = renderWithTheme(
      <ImageDrop dropzone={mockedDropzone} />
    );
    // Assert
    expect(mockedGetRootProps).toHaveBeenCalledWith({
      style: {
        padding: '32px',
        backgroundColor: 'blue',
        border: '1px dashed gray',
        color: 'gray',
        borderRadius: '4px',
      },
    });
    expect(mockedGetInputProps).toHaveBeenCalledWith();
    expect(baseElement).toMatchSnapshot();
  });

  describe('when drag is not accepted', () => {
    test('renders successfully', () => {
      // Act
      const mockedDropzone = {
        getRootProps: mockedGetRootProps,
        getInputProps: mockedGetInputProps,
        isDragAccept: false,
      } as unknown as DropzoneState;
      // Arange
      const {} = renderWithTheme(<ImageDrop dropzone={mockedDropzone} />);
      // Assert
      expect(mockedGetRootProps).toHaveBeenCalledWith({
        style: {
          padding: '32px',
          backgroundColor: '#fafafa',
          border: '1px dashed gray',
          color: 'gray',
          borderRadius: '4px',
        },
      });
      expect(mockedGetInputProps).toHaveBeenCalledWith();
    });
  });
});
