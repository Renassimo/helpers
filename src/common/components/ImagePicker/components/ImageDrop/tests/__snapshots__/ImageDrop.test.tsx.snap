// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageDrop renders snapshot successfully 1`] = `
<body>
  <div>
    <div
      class="mockedGetRootPropsResult"
    >
      <input
        type="input"
      />
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        Drag and drop map image, or click to select...
      </p>
    </div>
  </div>
</body>
`;
